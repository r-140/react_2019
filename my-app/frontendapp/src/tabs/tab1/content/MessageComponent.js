import React from 'react'

export default class MessageComponent extends React.Component {
	render() {
		console.log(this)
		return (
			<div>
				
					<h1>{this.props.msg}</h1>
				
			</div>
		)
	}
}