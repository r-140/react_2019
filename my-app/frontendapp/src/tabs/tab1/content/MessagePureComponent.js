import React from 'react'


export default class MessagePureComponent extends React.PureComponent {
	render() {
		console.log(this)
		return (
			<div>
				
					<h1>{this.props.msg}</h1>
				
			</div>
		)
	}
}