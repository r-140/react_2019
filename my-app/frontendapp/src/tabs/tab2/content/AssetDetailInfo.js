import React from 'react'

import { Link } from 'react-router-dom';

export class AssetDetailInfo extends React.Component {

	state = {
		asset: { name: '' }
		
	};


	componentDidMount(){
		const { match: { params } } = this.props;

		fetch(`http://localhost:63145/api/assets/asset/${params.id}`)
            .then(res => res.json())
            .then(
                (asset) => {
                    
                    this.setState({
                        asset: asset[0]
                    });
                },
                (error) => {
                    this.error = error;
                }
            )
	}
	render() {
		
		const {asset} = this.state;
		console.log("asset name ", asset);
		return (
			<div >
				<div> {asset.name}</div>	

				<div>
				<Link to="/assets">
						<button >Close</button >
					</Link>
				</div>
			</div>
		)
	}
	
}