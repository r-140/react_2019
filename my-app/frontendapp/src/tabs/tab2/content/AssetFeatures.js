import React from 'react'

import { Link } from 'react-router-dom';

export class AssetFeatures extends React.Component {

	state = {
		asset: { name: '' }
		
	};


	componentDidMount(){
		const { match: { params } } = this.props;
		console.log("params ", this.props)

		fetch(`http://localhost:63145/api/assets/asset/${params.id}`)
            .then(res => res.json())
            .then(
                (asset) => {
                    console.log("params ", params)
                    this.setState({
                        asset: asset[0]
                    });
                },
                (error) => {
					console.log("params ", params)
                    this.error = error;
                }
            )
	}
	render() {
		
		const {asset} = this.state;
		
		console.log("asset name ", asset);
		return (
			<div >
				<div>
					<div>ID: {asset.id}</div>

					<div>Name: {asset.name}</div>	

					<div>Domain: {asset.domain}</div>	

					<div>Domain2: {asset.domain2}</div>	

					<div>Author: {asset.created_by}</div>	
				</div>

				<div>
					<Link to="/assets">
							<button >Close</button >
					</Link>
				</div>
			</div>
		)
	}
	
}