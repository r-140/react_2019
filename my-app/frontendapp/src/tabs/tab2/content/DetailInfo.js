import React from 'react'

export default class AssetDetailInfo extends React.Component {
	render() {
		console.log(this)
		return (
			<div>				
                <div>{this.props.assetId}</div>				
			</div>
		)
	}
}