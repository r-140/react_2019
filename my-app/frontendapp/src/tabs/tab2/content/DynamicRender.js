import  React, {Fragment} from 'react';
import {connect} from "react-redux";
import {loadAssets} from "../../../reducers"

 class DynamicRender extends React.Component {

    state = {inputValue:""};
  componentDidMount() {
      // this.props.loadAssets(5000)
  }

  onSearch = () => {this.props.loadAssets(this.state.inputValue)};

  render() {
      const { assets} = this.props;
    // if (error) {
    //   return <div>Error: {error.message}</div>;
    // }
    // else if (!isLoaded) {
    //   return <div>Loading...</div>;
    // }
    // else {
      return (
        
       <ul>
      
           <input value={this.state.inputValue} onChange={(event) => this.setState({inputValue: event.target.value})}/>
           <button onClick={this.onSearch}>Search</button>
         {Object.keys(assets).map(key => (
           <li key={key}>
             {'Author:' + key + ' count: ' + assets[key]}
           </li>
         ))}
       </ul>

      );
    }
  // }
}

export default connect(state => {
    console.log(state);
    return state
}, {loadAssets})(DynamicRender)




