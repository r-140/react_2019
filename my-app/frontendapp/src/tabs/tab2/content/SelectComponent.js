import React from 'react'

import uuidv1 from "uuid";

export class SelectComponent extends React.Component {

	state = {
		defaultValue: null
	}
	

	render() {
		const { list, label } = this.props;
		console.log("list is empty ", isEmpty(list));
		return (
			<label>
				{label}
				<select value={this.props.value || ''} onChange={e => this.props.onChange(this.props.filterName, e.target.value)}>
				<option value=''></option>
				
				if(isEmpty(list)===false)
				{
					// <option value=''></option>
					list.map(item => (
						<option key={uuidv1()} value={item.pathid}>
						{item.name}
						</option>
					))
				}
				
				</select>
			</label>
		)
	}

	
	
}
var hasOwnProperty = Object.prototype.hasOwnProperty;
function isEmpty(obj) {

    // null and undefined are "empty"
    if (obj == null) return true;

    // Assume if it has a length property with a non-zero value
    // that that property is correct.
    if (obj.length > 0)    return false;
    if (obj.length === 0)  return true;

    // If it isn't an object at this point
    // it is empty, but it can't be anything *but* empty
    // Is it empty?  Depends on your application.
    if (typeof obj !== "object") return true;

    // Otherwise, does it have any properties of its own?
    // Note that this doesn't handle
    // toString and valueOf enumeration bugs in IE < 9
    for (var key in obj) {
        if (hasOwnProperty.call(obj, key)) return false;
    }

    return true;
}

SelectComponent.defaultProps = {
	list: {},
	filterName: 'filterName',
	label: 'select',
	onChange: () => {}
  };