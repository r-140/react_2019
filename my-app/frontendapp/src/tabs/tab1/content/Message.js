import React from 'react'

import  MessageComponent  from './MessageComponent';

import  MessagePureComponent  from './MessagePureComponent';

import  {MessageFunctionalComponent} from './MessageFunctionalComponent';



import  MessageElement  from './MessageElement';

export default class Message extends React.Component {
	render() {
		console.log(this)
		return (
			<div>
				<div>
					React create element example:
					<MessageElement msg = "Das ist meine erste React App"/>
					
				</div>
				
				<div>
					React create Component example
					<MessageComponent msg= "Das ist der Beispiel des ReactComponent "/>
					
				</div>

				<div>
					React create Pure Component example
					<MessagePureComponent msg= "Das ist der Beispiel des ReactPureCompoment . In Zukunft ich kann diesen Komponenten vergleichen"/>
					
				</div>

				<div>
					React create Functional Component example
					<MessageFunctionalComponent msg= "Das ist der Beispiel des React Functional Compoment . Es benutzt arrow func. Das idt wunderbar"/>
					
				</div>

				
			</div>
		)
	}
}
