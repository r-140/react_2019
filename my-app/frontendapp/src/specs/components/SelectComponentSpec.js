import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import {SelectComponent} from '../../tabs/tab2/content/SelectComponent';


const list = [
  {
    pathid: 'root.',
    name: 'Root',
  },
  {
    pathid: 'root.krone.',
    name: 'krone',
  },
];

const label = "label";

describe('<SelectComponent />', () => {
  let component;
  let tree;

 
  test('Should render passed list', () => {
    component = renderer.create(<SelectComponent label={label} list={list} />);
    tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should call onChange when changed', () => {
    const onChange = jest.fn();
    component = mount(<SelectComponent label={label} list={list} onChange={onChange}/>);

    component.find('select').simulate('change');
    // component.simulate('change');

    expect(onChange.mock.calls.length).toBe(1);
    expect(onChange.mock.calls[0][0]).toBe(list.item.pathid);
  });

});