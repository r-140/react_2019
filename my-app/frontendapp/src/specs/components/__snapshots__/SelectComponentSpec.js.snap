// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectComponent /> Should render passed list 1`] = `
<label>
  label
  <select
    onChange={[Function]}
    value=""
  >
    <option
      value=""
    />
    if(isEmpty(list)===false)
    <option
      value="root."
    >
      Root
    </option>
    <option
      value="root.krone."
    >
      krone
    </option>
  </select>
</label>
`;
